packages/component-library/src/components/form/mt-text-field/mt-text-field.interactive.stories.ts

```diff
@@ -230,3 +230,28 @@ export const VisualTestInheritanceActive: MtTextFieldStory = {
     isInherited: true,
   },
 };
+
+export const TestHandleFocus: MtTextFieldStory = {
+  name: "Should handle focus",
+  args: {
+    placeholder: "Focus here",
+  },
+  play: async ({ canvasElement, args }) => {
+    const canvas = within(canvasElement);
+    const input = canvas.getByPlaceholderText(args.placeholder);
+
+    // Listen for the focus event
+    const focusEvent = new Promise((resolve) => {
+      input.addEventListener("focus", resolve);
+    });
+
+    // Simulate focus event
+    await fireEvent.focus(input);
```

Comment:
This doesn't test the real handleFocus behavior of the component. The test creates a event listener on it's own and triggers it directly. Then it checks afterwards if a Promise is defined (which is the case everytime).

Better is to test the real implementation:
1. User clicks on the input
2. Check if the component emits the "focus" event (you can check this directly in Storybook testing with args, see https://github.com/shopware/meteor/blob/main/packages/component-library/src/components/table-and-list/mt-data-table/mt-data-table.interactive.stories.ts#L174 )

Also it would be useful to change the name to `VisualTestHandleFocus` so that it generates a snapshot. Then the focus class is also tested